### PR TITLE
Added a test framework

### PR DESCRIPTION
**Pull request by gricey432**
I was trying to get testing working with async but @TRManderson gave the impression that what I was trying to do isn't possible. So I've done this instead.

Not super happy with it, and very open to suggestions.

Also fixed a bunch of PEP8 stuff while I was there.

Progresses on #254

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1 +0,0 @@
--e .
```
Comment (gricey432) [inline requirements.txt]: This causes issues with Pycharm's detection of requirements

**Comment by TRManderson** [comment]
Just a general comment - this isn't Polymathian. UQCSBot doesn't necessarily follow the Polymathian style guide, and "Because PyCharm" isn't necessarily a good enough reason like it is at work.

**Comment by mitchmcdee** [comment]
I'm all for the style guide changes, but it heavily pollutes the PR-- could it be separated into its own PR?

**Comment by gricey432** [comment]
@TRManderson all of the style changes are PEP8 apart from the `requirements.txt`. I'd say PEP8 should apply everywhere.
As for the `requirements.txt` file, it was more a combo of "it messes with pycharm" and "there's no documented reason this needs to exist". Idno what it's for, but I can undo that part if it actually has a use.

@mitchmcdee how keen are you on that? I did all this in one commit, so I'd have to split the changes by hand which would suck a bit

**Comment by mitchmcdee** [comment]
well they're just style changes with no dependencies-- should be easy to just copy paste each file into a new branch and then `git checkout -- <file>`. Would just make reviewing a heck of a lot easier :^)

**Comment by gricey432** [comment]
@mitchmcdee after many screw ups with interactive rebasing, I've managed to split it into two branches. This one only has the test stuff now.

**Comment by mitchmcdee** [comment]
kek 😂 coulda just done it in a new commit!

**Comment by mitchmcdee** [comment]
Could you add a little blurb to the README explaining how to run tests?

**Comment by gricey432** [comment]
Added readme

File: README.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ If a bot was available, it will now be running on uqcstesting.
 ## Running the bot in a custom Slack team
 
 1. [Create a Slack workspace](https://slack.com/create)
-2. Run `python3 setup.py install`
+2. Run `pip install -e .`
```
Comment (mitchmcdee) [inline README.md]: this is my bad for forgetting to remove step 2 when I re-added it, but you can probably omit this line entirely seeing as though its stated in 'Setting up the bot'

**Pull request merged**